### PR TITLE
[New Hack] 066-Open AI Fundamentals

### PR DESCRIPTION
**Pull request by raccchel**

### CODE REVIEW

File: 066-OpenAIFundamentals/Student/Challenge-01.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Model deployment for the challenge:
+- Deploy the following models in your Azure OpenAI resource. 
+  - `gpt-35-turbo`
+  - `text-ada-001`
+  - `text-babbage-001`
+  - `text-curie-001`
+  - `text-davinci-003`
+  - `text-embedding-ada-002`
+    
+(!) Model families currently available as of _Aug 4, 2023_ in Azure OpenAI includes GPT-3, Codex and Embeddings, GPT-4 is available for application. Please reference this link for more information: https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models.
```
Comment (perktime) [inline 066-OpenAIFundamentals/Student/Challenge-01.md:28]: What is the (!) for? Should it say "Note: "?

**Review by perktime — CHANGES_REQUESTED** [review]

File: 066-OpenAIFundamentals/Coach/Solution-00.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Challenge 00 - <Title of Challenge> - Coach's Guide
```
[the remark this answers is not in the document]
Comment (devanshithakar12) [reply 066-OpenAIFundamentals/Coach/Solution-00.md]: Fixed

File: 066-OpenAIFundamentals/Coach/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+- [Azure subscription](https://portal.azure.com/) - Estimated spend may be around $10 based on running Cognitive Search for four days (total length of time depends on implementation time)
+- [Access to Azure OpenAI](https://customervoice.microsoft.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR7en2Ais5pxKtso_Pz4b1_xUOFA5Qk1UWDRBMjg0WFhPMkIzTzhKQ1dWNyQlQCN0PWcu)
+- Jupyter Notebook editor (we recommend [Visual Studio Code](https://code.visualstudio.com/Download) or [Azure Machine Learning Studio](https://ml.azure.com/))
+	- If using Visual Studio Code, we also recommend installing [Anaconda](https://docs.anaconda.com/anaconda/install) OR [Miniconda](https://docs.anaconda.com/anaconda/install) for project environment management
+- [Python](https://www.python.org/downloads/) (version 3.7.1 or later), plus the package installer [pip](https://pypi.org/project/pip/)
+- [Azure Cognitive Search](https://learn.microsoft.com/azure/search) (Basic Tier) - This will be created during the Hack and is not necessary to get started.
+
+## Suggested Hack Agenda (Optional)
```
[the remark this answers is not in the document]
Comment (devanshithakar12) [reply 066-OpenAIFundamentals/Coach/README.md]: Removed

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-03-B-Chunking.ipynb [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+    "def count_tokens(string: str, encoding_name: str) -> int:\n",
+    "    # Enter code here\n",
+    "    return "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Student Challenge #2:\n",
```
Comment (perktime) [inline 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-03-B-Chunking.ipynb:125]: Note: this comment applies generally to this file. Within this challenge, there are multiple challenges which is a bit confusing. Perhaps the challenges within this file should be named "tasks" instead.

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-04-A-RAG_for_structured_data.ipynb [hunk 5]
```diff
@@ -0,0 +1,1124 @@
+{
```
Comment (perktime) [inline 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-04-A-RAG_for_structured_data.ipynb:1]: Might be better to stay away from potentially sensitive news topics for this challenge (i.e., politics).

File: 066-OpenAIFundamentals/Coach/Solutions/CH-02-ModelComparison-Solution.ipynb [hunk 6]
```diff
@@ -0,0 +1,1748 @@
...
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "\n",
+        "[Azure OpenAI models](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/concepts/models)  \n",
+        "\n",
+        "\n",
+        "| | Similarity embedding | Text search embedding | Code Search Embedding |\n",
```
Comment (perktime) [inline 066-OpenAIFundamentals/Coach/Solutions/CH-02-ModelComparison-Solution.ipynb:124]: This section looks strange in VS Code

**Review by perktime — DISMISSED** [review]

File: 066-OpenAIFundamentals/Coach/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Coach/README.md]: I think this needs to be RE-ADDED and have some content in it!

Add "Suggested Hack Agenda" section:
□ Include guidance for hosting this hack vs. directing students to self-administer
□ Suggest 2-week async format? Describe async format
	® synchronous kick-off & office hour calls
	® Asynchronous support via Microsoft Teams, one channel per challenge that the SMEs/Coaches monitor for the duration of the hack
□ Run it the traditional way 2-3 days w/Coaches?
□ Any notes on the order of the challenges (iterative, vs choose your own adventure.) For example, C2 is optional?

**Comment by raccchel** [comment]
> @raccchel please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by raccchel** [comment]
> @raccchel please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: 066-OpenAIFundamentals/Coach/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
+# What The Hack - Azure OpenAI Fundamentals - Coach Guide
+
+## Introduction
+
+Welcome to the coach's guide for the Azure OpenAI Fundamentals What The Hack. While this Hack is designed to be taken individually in a self-paced, self-administered manner, here you will find links to specific guidance for each of the challenges. There can be multiple ways to implement a solution in the challenges, so these solution guides are non-exhaustive but are rather, guides for when participants are feeling stuck or if they would like to compare solutions.
+
+**NOTE:** If you are a Hackathon participant, this is the answer guide. We encourage you to challenge yourself and avoid cheating yourself out of the experience by only referencing these when needed.
+
+
+This hack includes an optional [lecture presentation](Lectures.pptx) that features short presentations to introduce key topics associated with each challenge.
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Coach/README.md]: Commented this out for now.  Will uncomment this if/when they do publish a PPT

File: 066-OpenAIFundamentals/Coach/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Coach/README.md]: Edits made.  Resolving this thread.

**Review by jrzyshr — CHANGES_REQUESTED** [review]

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-02-ModelComparison.ipynb [hunk 8]
```diff
@@ -0,0 +1,1264 @@
...
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "\n",
+        "[Azure OpenAI models](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/concepts/models)  \n",
+        "\n",
+        "\n",
+        "| | Similarity embedding | Text search embedding | Code Search Embedding |\n",
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-02-ModelComparison.ipynb:105]: Fixed!  It was missing a spacer in the column definition.

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-04-A-RAG_for_structured_data.ipynb (hunk 5 above)
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-04-A-RAG_for_structured_data.ipynb:1]: Feedback noted. The authors will consider a different data source in vNext of this hack.

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-02-ModelComparison.ipynb [hunk 9]
```diff
@@ -0,0 +1,1264 @@
...
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "### 1.5 Quotas and Limits\n",
+        "\n",
+        "*The below limits are subject to change. We anticipate that you will need higher limits as you move toward production and your solution scales. When you know your solution requirements, please reach out to us by applying for a quota increase here: https://aka.ms/oai/quotaincrease\n",
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-02-ModelComparison.ipynb]: Text fixed.

File: 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-01-PromptEngineering.ipynb [hunk 10]
```diff
@@ -0,0 +1,1981 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "#### Create the environment file \".env\"\n",
```
[the remark this answers is not in the document]
Comment (jrzyshr) [reply 066-OpenAIFundamentals/Student/Resources/Notebooks/CH-01-PromptEngineering.ipynb]: Updated text to clarify.  Jupyter notebook can be loaded before an ENV file is in place, and you can update the ENV file at any time.

**Review by jrzyshr — APPROVED** [review]
Thanks team for the contribution of our FIRST hack on OpenAI!!!!!

**Pull request merged**